sys/net/gnrc/netreg/gnrc_netreg.c

```diff
@@ -39,21 +39,20 @@ void gnrc_netreg_init(void)
 
 int gnrc_netreg_register(gnrc_nettype_t type, gnrc_netreg_entry_t *entry)
 {
-#ifdef DEVELHELP
-#if defined(MODULE_GNRC_NETAPI_MBOX) || defined(MODULE_GNRC_NETAPI_CALLBACKS)
-    bool has_msg_q = (entry->type != GNRC_NETREG_TYPE_DEFAULT) ||
-                     sched_threads[entry->target.pid]->msg_array;
-#else
-    bool has_msg_q = sched_threads[entry->target.pid]->msg_array;
-#endif
+#if DEVELHELP
+    bool has_msg_q = thread_has_queue(sched_threads[entry->target.pid]);
+
+# if defined(MODULE_GNRC_NETAPI_MBOX) || defined(MODULE_GNRC_NETAPI_CALLBACKS)
+    has_msg_q = (entry->type != GNRC_NETREG_TYPE_DEFAULT) || has_msg_q;
```

Comment:
It's less readable, but if you move the ```thread_has_queue()``` call after the ||, it'll be executed conditionally.

Reply:
For `DEVELHELP` code, I rather have it readable than optimized.

Reply:
ok!